### ISSUE TITLE
Support "disabled by default" setting / method

### ISSUE THREAD

**Issue by JohnMcLear** [ether/ep_webrtc#23]
This is different from passing AV=no in URI as it still displays other peoples stream but it means you don't auto broadcast your video/audio to recipients by default instead opting in to turn on your mic / video

**Comment by seballot** [comment]
Hi ! I would like to work on this one if it's okay :)

**Comment by seballot** [comment]
Mmm, I had a look, but not so simple, need to understand RTC protocol etc...
I'm running a bit out of time re my contributions to etherpad, so sorry I give up with this one

**Comment by JohnMcLear** [comment]
Don't worry @seballot I will get onto this once 1.8.1 is ready

**Comment by orblivion** [comment]
I'll give this a shot, seems simple. It sounds like you want a single setting for both audio and video?

**Comment by JohnMcLear** [comment]
Yea I think that's the goal!  

I guess in general all settings in ep_webrtc need porting to

settings.json.template type thing -- so a modification to the file
A setting within the client editor (stored as cookie).
A setting passable w/ URL parameters

Think

```
camera: default / on / off,
mic: default / on / off,
disable: default / on / off
```

A site admin might want to leave Audio chat only turned on for their instance for example..

**Comment by orblivion** [comment]
Okay, sounding slightly less simple :-)

I'm having a little trouble parsing this. Could you clarify:

>  A site admin might want to leave Audio chat only turned on for their instance for example

Do you mean _disabling_ video instance-wide, or just having video off by default?

**Comment by JohnMcLear** [comment]
Okay lemme break it down..

1. Settings.json - instance wide, can be [forced/default]
1. User - user settings, changed by clicking the settings cog > saves as cookies.
1. User - user settings, changed by appending a parameter to the URI IE ``/p/foo?av=true``

Whenever we introduce a new setting for any plugin we should think to if it makes sense to expose those 3.  The thought process is.

If we take "no video feature request" as an example we go through this process..

1. Might a site admin want a video only site?  
1. Yes.  Add settings.json, might a user want it?  
1. Yes.  Might a user want a URL parameter.

So it makes sense to just bring in the logic for all 3 options to change the settings at one time, that way you can extend all settings as one big job using the same code.

**Comment by JohnMcLear** [comment]
It's also worth noting (and this is somewhat out of scope) that site admins often want to override language strings IE error msgs so brace to bring that in too :)

**Comment by orblivion** [comment]
Okay, I think that sort of clears it up. Practically speaking though, it seems like it would have to be laid out this way:

In `settings.json` (site admin level), the values for a given field could be:

* default on (implies enabled)
* default off (implies enabled)
* disabled

And then for each user, assuming the admin didn't pick "disabled", their choices (in url and in settings cog) are either:

* default on
* default off

Dealing with legacy will be a bit confusing since we have something called `ep_webrtc.enabled` for A/V as a whole, which doesn't actually function like "enabled" vs "disabled". Rather, it functions like "default on" vs "default off".

Anyway, I think I have enough to get started, and we can hash out further misunderstandings in PR.

---

Custom error messages seem easy to me.

**Comment by JohnMcLear** [comment]
Awesome tnx :)

**Comment by orblivion** [comment]
Done via https://github.com/ether/ep_webrtc/pull/43 ?
